src/amp-story-ad-preview.js

```diff
@@ -129,4 +162,38 @@ export default class AmpStoryAdPreview {
     // b) reject when invalid
     this.writeToIframe_(await this.adIframe_, patch(dirty));
   }
+
+  async setMetaCtaLabel_(dirty) {
+    const {ctaType, ctaUrl} = this.extractMetaCta_(dirty);
+    const storyCta = (await this.storyIframe_).contentDocument.querySelector(
+      '.i-amphtml-story-ad-link'
+    );
+    storyCta.textContent = CTA_TYPES[ctaType];
+    storyCta.setAttribute('href', ctaUrl);
+  }
+
+  extractMetaCta_(dirty) {
+    const head = dirty.substring(0, dirty.lastIndexOf('</head>'));
+    const components = head.split('<');
+    let ctaType = defaultCtaType;
+    let ctaUrl = defaultCtaUrl;
+    for (let component of components) {
+      if (component.includes(`meta name="amp-cta-type"`)) {
+        ctaType = component.substring(
+          component.lastIndexOf('=') + 2,
+          component.lastIndexOf(`"`)
+        );
+      }
+      if (component.includes(`meta name="amp-cta-url"`)) {
+        ctaUrl = component.substring(
+          component.lastIndexOf('=') + 2,
+          component.lastIndexOf(`"`)
+        );
+      }
+    }
+    return {
```

Comment:
nit: since the keys and value identifiers are the same, you can use modern shorthand syntax `{ctaType, ctaUrl}`